Sources/WasmKitGDBHandler/DebuggerMemoryCache.swift

```diff
@@ -0,0 +1,57 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#if WasmDebuggingSupport
+
+    import NIOCore
+    import WasmKit
+
+    package struct DebuggerMemoryCache: ~Copyable {
```

Comment:
This type no longer has any state or cache, so let's call it a `DebuggerMemoryView`.